pm_nq/pm_nq.xml

```diff
@@ -0,0 +1,253 @@
+            </when>
+            <when value="r">
+                <param name="zipped_input" type="data" format="zip" label="Zipped Input" help="A zipped directory structure containing the output of the generate mode, and the results of running CASTEP or DFTB+ to generate the desired property."/>
+            </when>
+        </conditional>
+        <param type="hidden" name="testing" label="Test mode" value="false"/>
+    </inputs>
+    <outputs>
+        <data label="Displaced Structures" name="displaced_structures" format="zip" from_work_dir="out_zip.zip">
+            <filter>(task['task'] == 'w')</filter>
+        </data>
+        <data label="File tree (testing only)" name="file_tree" format="txt" from_work_dir="tree.txt" hidden="true">
+            <filter>(task['task'] == 'w' and testing == 'true')</filter>
+        </data>
+        <data label="Averaged Property" name="averaged_property" format="txt" from_work_dir="averages.dat">
+            <filter>(task['task'] == 'r')</filter>
+        </data>
+    </outputs>
+    <tests>
+        <!-- Test CASTEP without default values -->
+        <test expect_num_outputs="2">
+            <param name="testing" value="true"/>
+            <param name="structure" value="Si.cell" ftype="cell"/>
+            <conditional name="task">
+                <param name="task" value="w"/>
+                <conditional name="phonon">
+                    <param name="phonon_source_castep" value="Si.phonon" ftype="txt"/>
+                    <param name="castep_file" value="Si.castep"/>
+                </conditional>
+            </conditional>
+            <output name="file_tree" file="tree.txt" ftype="txt" sort="true"/>
+        </test>
+        <!-- Test CASTEP with optional extra files -->
+        <test expect_num_outputs="2">
+            <param name="testing" value="true"/>
+            <param name="structure" value="Si.cell" ftype="cell"/>
+            <conditional name="task">
+                <param name="task" value="w"/>
+                <param name="script_file" value="submit.sh" ftype="txt"/>
+                <conditional name="phonon">
+                    <param name="phonon_source_castep" value="Si.phonon" ftype="txt"/>
+                    <param name="castep_file" value="Si.castep"/>
+                </conditional>
+                <conditional name="calculator">
+                    <param name="castep_param" value="Si.param"/>
+                </conditional>
+            </conditional>
+            <output name="file_tree" file="tree_extra_files.txt" ftype="txt" sort="true"/>
+        </test>
+        <!-- Test DFTB+ -->
+        <test expect_num_outputs="2">
+            <param name="testing" value="true"/>
+            <param name="structure" value="ethyleneMu_opt.xyz" ftype="xyz"/>
+            <conditional name="task">
+                <param name="task" value="w"/>
+                <param name="method" value="montecarlo"/>
```

Comment:
No the choice of sampling method and calculator are independent, I just changed both at the same time so I could hit both with one test case (i.e. because I'm lazy).